src/parser.ts

```diff
@@ -4445,13 +4457,99 @@ export function parseImportExpression(
   if (parser.getToken() === Token.Ellipsis) report(parser, Errors.InvalidSpreadInImport);
 
   const source = parseExpression(parser, context, 1, inGroup, parser.tokenPos, parser.linePos, parser.colPos);
+  const node: ESTree.ImportExpression = {
+    type: 'ImportExpression',
+    source
+  };
+
+  if (context & Context.OptionsNext) {
+    let options: ESTree.Expression | null = null;
+
+    if (parser.getToken() === Token.Comma) {
+      consume(parser, context, Token.Comma);
+
+      if (parser.getToken() !== Token.RightParen) {
+        const expContext = (context | Context.DisallowIn) ^ Context.DisallowIn;
+        options = parseExpression(parser, expContext, 1, inGroup, parser.tokenPos, parser.linePos, parser.colPos);
+      }
+    }
+
+    node.options = options;
+    consumeOpt(parser, context, Token.Comma);
+  }
 
   consume(parser, context, Token.RightParen);
 
-  return finishNode(parser, context, start, line, column, {
-    type: 'ImportExpression',
-    source
-  });
+  return finishNode(parser, context, start, line, column, node);
+}
+
+/**
+ * Parses import attributes
+ *
+ * @param parser Parser object
+ * @param context Context masks
+ * @returns
+ */
+
+export function parseImportAttributes(
+  parser: ParserState,
+  context: Context,
+  specifiers: ESTree.ImportDeclaration['specifiers'] = []
+): ESTree.ImportAttribute[] {
+  consume(parser, context, Token.WithKeyword);
+  consume(parser, context, Token.LeftBrace);
+
+  const attributes: ESTree.ImportAttribute[] = [];
+  const keysContent = new Set<ESTree.Literal['value'] | ESTree.Identifier['name']>();
+  const validJSONImportAttributeBindings =
+    specifiers.length >= 1 &&
+    (specifiers[0].type === 'ImportDefaultSpecifier' ||
+      specifiers[0].type === 'ImportNamespaceSpecifier' ||
+      (specifiers[0].type === 'ImportSpecifier' && specifiers[0].imported.name === 'default'));
+
+  while (parser.getToken() !== Token.RightBrace) {
+    const start = parser.tokenPos;
+    const line = parser.linePos;
+    const column = parser.colPos;
+
+    const key = parseIdentifierOrLiteral(parser, context);
+    consume(parser, context, Token.Colon);
+    const value = parseLiteral(parser, context);
+    const keyContent = key.type === 'Literal' ? key.value : key.name;
+    const isJSONImportAttribute = keyContent === 'type' && value.value === 'json';
```

Comment:
Why is this needed? The value of type shouldn't matters at all.

Reply:
Ah, because we support `json-modules` feature? https://github.com/tc39/proposal-json-modules#why-dont-json-modules-support-named-exports

Reply:
@fisker this is to fix this test: `test262/test/language/import/import-attributes/json-named-bindings.js`. As it says:

```
In the early design of JSON modules, contributors considered allowing the
  properties of object values in JSON modules to be imported directly by name.
  This was ultimately rejected, so attempting to import in this way should
  produce a SyntaxError.
```

However, this is specific only to JSON. For other import attributes, such as Bun macros, it has to be done for named bindings: https://bun.sh/docs/bundler/macros

Reply:
I don't think any parser rejects this now, maybe we can put `json-modules` into [unsupported features](https://github.com/meriyah/meriyah/blob/master/test/test262-parser-tests/unsupported-features.txt). But lets hear what @3cp thinks first.

Reply:
No problem, or I imagine it will be putting this `test262/test/language/import/import-attributes/json-named-bindings.js` to the whitelist. Well, I'll wait for @3cp's answer and see what he thinks. Thanks!

Reply:
I wish tc39 pushed this as runtime error, not syntax error :-)